### ISSUE TITLE
Learning rate finder wrong error message on early stopping

### ISSUE THREAD

**Issue by fg91** [PyTorchLightning/pytorch-lightning#5044]
If the learning rate finder stops early, an error is shown that 'LR finder stopped early due to diverging loss.'.

**However, early stopping can also happen if the data loader is shorter than the default number of training steps (100) performed by the learning rate finder. In this case, the error message about diverging loss is truly misleading.** 

https://github.com/PyTorchLightning/pytorch-lightning/blob/20b806a94552a24aaf82861bf9b021674ef4bcf0/pytorch_lightning/tuner/lr_finder.py#L185

```python
  # Prompt if we stopped early
    if trainer.global_step != num_training:
        log.info('LR finder stopped early due to diverging loss.')
```

**Comment by fg91** [comment]
Not stale

**Comment by edenlightning** [comment]
mind sending a PR to fix?

**Comment by fg91** [comment]
> mind sending a PR to fix?

Will do 👍

**Comment by edenlightning** [comment]
Hey @fg91! Any way we can help you?

**Comment by fg91** [comment]
> Hey @fg91! Any way we can help you?

While creating a minimal working example to reproduce the issue and then fix it, I discovered, that when the length of the data loader used by the learning rate finder is smaller than the number of training steps, it appears to repeat over the same dataloader again. This means that the issue could be closed now.

Only thing that would make sense to me is to also check that the loss that was reached by the trainer actually is larger than `early_stop_threshold` before reporting `log.info('LR finder stopped early due to diverging loss.')`. Currently this is assumed but not checked. Would you like me to add that check?

**Comment by cowwoc** [comment]
@edenlightning I think @fg91 is waiting for your reply

**Comment by tchaton** [comment]
Thanks @fg91 @cowwoc.

Closing this issue as the behaviour seems accurate as we are re-iterating over the dataloader.

Best,
T.C
